### PR TITLE
Support to get Action from git, in order to debug Action quickly 

### PR DESCRIPTION
**Pull request by ZLValien**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

Modified the code of the DiceHub module

Support to get pipeline-action from git
Support to concurrent query extension 

pipeline get  pipeline-action definition form erda-action project before. In order to debug pipeline-action ,support get it form git

because get pipeline-action form git need a lot of time. Support concurrent query. Call the interface once for about 5 seconds

### CODE REVIEW

**Comment by sfwn** [comment]
Give us a sample of pipeline.yml.

**Comment by sfwn** [comment]
How about the speed? We recommend use app gittar url, not github. Test it please.

**Comment by ZLValien** [comment]
> How about the speed? We recommend use app gittar url, not github. Test it please.
local:
https://github.com/ZLValien/erda-actions    10s
https://erda.hkci.terminus.io/integration/dop/zzzz/action  2s

**Comment by Effet** [comment]
if not related about github, please amend your pr title @ZLValien

**Comment by ZLValien** [comment]
this can be tested
POST http://openapi.dev.terminus.io/api/extensions/actions/search
body:{
    "yamlFormat":false,
    "extensions":["test2@https://erda.hkci.terminus.io/integration/dop/zzzz/action"]
}

File: modules/dicehub/service/extension/git_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package extension
+
+//import (
+//	"gotest.tools/assert"
+//	"io/ioutil"
+//	"os"
+//	"testing"
+//)
+//
+//func Test_pullgit(t *testing.T) {
```
Comment (kakj-go) [inline modules/dicehub/service/extension/git_test.go:23]: if can not test getGit func, remove this test file

File: modules/dicehub/service/extension/git.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	// dirName is "uuid-xxxx"
+	uuid = uuid2.UUID()
+	dir, err := ioutil.TempDir(os.TempDir(), uuid+"-*")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(dir)
+
+	// git init
+	command := exec.Command("sh", "-c", "git init")
```
Comment (kakj-go) [inline modules/dicehub/service/extension/git.go:61]: git init should after cd temp dir

File: modules/dicehub/service/extension/git.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		Swagger:   "",
+		Readme:    files[2],
+		CreatedAt: time.Now(),
+		UpdatedAt: time.Now(),
+		IsDefault: false,
+		Public:    true,
+	}, nil
+}
+
+func getGit(d string, file ...string) ([]string, error) {
```
Comment (kakj-go) [inline modules/dicehub/service/extension/git.go:47]: change func name like `getGitFileContents`

**Comment by Effet** [comment]
still bad title, please explain what exactly you did.

**Comment by ZLValien** [comment]
https://erda.dev.terminus.io/terminus/dop/projects/2/apps/117/repo/tree/feature/sonar/pipeline.yml

test

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dicehub/service/extension/extension.go [hunk 4]
```diff
@@ -88,27 +89,49 @@ func (i *Extension) Create(req *apistructs.ExtensionCreateRequest) (*apistructs.
...
-			version = splits[1]
-		}
-		if version == "" {
-			extVersion, _ := i.GetExtensionDefaultVersion(name, req.YamlFormat)
-			result[fullName] = extVersion
-		} else {
-			extensionVersion, err := i.GetExtensionVersion(name, version, req.YamlFormat)
-			if err != nil {
-				result[fullName] = nil
+		go func(fnm string) {
```
[the remark this answers is not in the document]
Comment (ZLValien) [reply modules/dicehub/service/extension/extension.go:99]: Concurrently pull 100 small warehouses (10KB each), the test is successful

Concurrently pull 30 large warehouses (100MB each), the interface is timed out, the “defer” logic is not entered, so the folder is not deleted, the test is fail

**Review by sfwn — APPROVED** [review]

**Pull request merged**